### ISSUE TITLE
Improve docs and logging when consul is not used

### ISSUE THREAD

**Issue by cdrage** [hashicorp/nomad#552]
Really odd error when deploying the init redis example.

I'm using the latest master of Nomad within a Docker container.

__EDIT:__ this error occurs in both a docker container and bare-metal. with both, I'm using -network-interface wlan0.

tldr before you read below; consul is built-in to nomad rightttttt? I don't need to deploy my own in order for it to work, yes? this previously worked before on 0.1

```
    Loaded configuration from 
==> Starting Nomad agent...
2015/12/09 21:59:19 [DEBUG] fingerprint.env_aws: Error querying AWS Metadata URL, skipping
==> Nomad agent configuration:

                 Atlas: <disabled>
                Client: true
             Log Level: DEBUG
                Region: global (DC: dc1)
                Server: true

==> Nomad agent started! Log data will stream in below:

    2015/12/09 21:59:17 [INFO] raft: Node at 127.0.0.1:4647 [Follower] entering Follower state
    2015/12/09 21:59:17 [INFO] serf: EventMemberJoin: wikus.global 127.0.0.1
    2015/12/09 21:59:17 [INFO] nomad: starting 4 scheduling worker(s) for [service batch system _core]
    2015/12/09 21:59:17 [INFO] client: using state directory /tmp/NomadClient630021836
    2015/12/09 21:59:17 [INFO] client: using alloc directory /tmp/NomadClient499710907
    2015/12/09 21:59:20 [DEBUG] client: node registration complete
    2015/12/09 21:59:17 [INFO] nomad: adding server wikus.global (Addr: 127.0.0.1:4647) (DC: dc1)
    2015/12/09 21:59:17 [DEBUG] client: periodically fingerprinting consul at duration 15s
    2015/12/09 21:59:18 [WARN] raft: Heartbeat timeout reached, starting election
    2015/12/09 21:59:18 [INFO] raft: Node at 127.0.0.1:4647 [Candidate] entering Candidate state
    2015/12/09 21:59:18 [DEBUG] raft: Votes needed: 1
    2015/12/09 21:59:18 [DEBUG] raft: Vote granted. Tally: 1
    2015/12/09 21:59:18 [INFO] raft: Election won. Tally: 1
    2015/12/09 21:59:18 [INFO] raft: Node at 127.0.0.1:4647 [Leader] entering Leader state
    2015/12/09 21:59:18 [INFO] raft: Disabling EnableSingleNode (bootstrap)
    2015/12/09 21:59:18 [DEBUG] raft: Node 127.0.0.1:4647 updated peer set (2): [127.0.0.1:4647]
    2015/12/09 21:59:18 [INFO] nomad: cluster leadership acquired
    2015/12/09 21:59:20 [DEBUG] fingerprint.env_gce: Error querying GCE Metadata URL, skipping
    2015/12/09 21:59:20 [DEBUG] fingerprint.network: Detected interface wlan0  with IP 192.168.1.161 during fingerprinting
    2015/12/09 21:59:20 [WARN] fingerprint.network: Unable to read link speed from /sys/class/net/wlan0/speed
    2015/12/09 21:59:20 [DEBUG] fingerprint.network: Unable to read link speed; setting to default 100
    2015/12/09 21:59:20 [DEBUG] client: applied fingerprints [arch cpu host memory network storage]
    2015/12/09 21:59:20 [DEBUG] driver.docker: using client connection initialized from environment
    2015/12/09 21:59:20 [DEBUG] driver.docker: privileged containers are disabled
    2015/12/09 21:59:20 [DEBUG] client: available drivers [docker exec]
    2015/12/09 21:59:20 [DEBUG] client: updated allocations at index 1 (0 allocs)
    2015/12/09 21:59:20 [DEBUG] client: state updated to ready
    2015/12/09 21:59:20 [DEBUG] client: allocs: (added 0) (removed 0) (updated 0) (ignore 0)
    2015/12/09 22:00:00 [DEBUG] worker: dequeued evaluation 5f6dec29-ccdd-8870-bfce-5b076266a1b8
    2015/12/09 22:00:00 [DEBUG] http: Request /v1/jobs (981.437µs)
    2015/12/09 22:00:00 [DEBUG] sched: <Eval '5f6dec29-ccdd-8870-bfce-5b076266a1b8' JobID: 'example'>: allocs: (place 1) (update 0) (migrate 0) (stop 0) (ignore 0)
    2015/12/09 22:00:00 [DEBUG] worker: submitted plan for evaluation 5f6dec29-ccdd-8870-bfce-5b076266a1b8
    2015/12/09 22:00:00 [DEBUG] sched: <Eval '5f6dec29-ccdd-8870-bfce-5b076266a1b8' JobID: 'example'>: setting status to complete
    2015/12/09 22:00:00 [DEBUG] client: updated allocations at index 7 (1 allocs)
    2015/12/09 22:00:00 [DEBUG] client: allocs: (added 1) (removed 0) (updated 0) (ignore 0)
    2015/12/09 22:00:00 [DEBUG] worker: updated evaluation <Eval '5f6dec29-ccdd-8870-bfce-5b076266a1b8' JobID: 'example'>
    2015/12/09 22:00:00 [DEBUG] client: starting runner for alloc 'a832123c-17a8-d041-afcb-aeb29687634b'
    2015/12/09 22:00:00 [DEBUG] worker: ack for evaluation 5f6dec29-ccdd-8870-bfce-5b076266a1b8
    2015/12/09 22:00:00 [DEBUG] client: starting task context for 'redis' (alloc 'a832123c-17a8-d041-afcb-aeb29687634b')
    2015/12/09 22:00:00 [DEBUG] http: Request /v1/evaluation/5f6dec29-ccdd-8870-bfce-5b076266a1b8 (172.768µs)    2015/12/09 22:00:00 [DEBUG] driver.docker: using client connection initialized from environment

    2015/12/09 22:00:00 [DEBUG] http: Request /v1/evaluation/5f6dec29-ccdd-8870-bfce-5b076266a1b8/allocations (232.846µs)
    2015/12/09 22:01:16 [DEBUG] driver.docker: docker pull redis:latest succeeded
    2015/12/09 22:01:16 [DEBUG] driver.docker: identified image redis:latest as 0643f0c7d17d348881b2dab88cfdb57ad86dcf32ea22bd0b2cd0957b6c7ab8c3
    2015/12/09 22:01:16 [DEBUG] driver.docker: using 268435456 bytes memory for redis:latest
    2015/12/09 22:01:16 [DEBUG] driver.docker: using 500 cpu shares for redis:latest
    2015/12/09 22:01:16 [DEBUG] driver.docker: binding directories []string{"/tmp/NomadClient499710907/a832123c-17a8-d041-afcb-aeb29687634b/alloc:/alloc:rw,z", "/tmp/NomadClient499710907/a832123c-17a8-d041-afcb-aeb29687634b/redis:/local:rw,Z"} for redis:latest
    2015/12/09 22:01:16 [DEBUG] driver.docker: networking mode not specified; defaulting to bridge
    2015/12/09 22:01:16 [DEBUG] driver.docker: allocated port 192.168.1.161:58957 -> 6379 (mapped)
    2015/12/09 22:01:16 [DEBUG] driver.docker: exposed port 6379
    2015/12/09 22:01:16 [DEBUG] driver.docker: setting container name to: redis-a832123c-17a8-d041-afcb-aeb29687634b
    2015/12/09 22:01:16 [INFO] driver.docker: created container 6823e10417490fa25136c2bd21b5c92a0849309bf2f3008b592d6e36f7d0c9a8
    2015/12/09 22:01:16 [INFO] driver.docker: started container 6823e10417490fa25136c2bd21b5c92a0849309bf2f3008b592d6e36f7d0c9a8
    2015/12/09 22:01:16 [INFO] consul: Registering service cache-redis with consul.
    2015/12/09 22:01:16 [DEBUG] consul: Error while registering service cache-redis with consul: Put http://127.0.0.1:8500/v1/agent/service/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
    2015/12/09 22:01:16 [DEBUG] consul: Registering Check with ID: e7a0a5ece0c8424b3c8b06248040a331d999558b for Service: a832123c-17a8-d041-afcb-aeb29687634b-cache-redis
    2015/12/09 22:01:16 [DEBUG] client: updated allocations at index 10 (1 allocs)
    2015/12/09 22:01:16 [DEBUG] client: allocs: (added 0) (removed 0) (updated 1) (ignore 0)
    2015/12/09 22:01:16 [ERROR] consul: Error while registerting check alive with consul: Put http://127.0.0.1:8500/v1/agent/check/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
    2015/12/09 22:01:20 [DEBUG] consul: Error while registering service cache-redis with consul: Put http://127.0.0.1:8500/v1/agent/service/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
    2015/12/09 22:01:20 [DEBUG] consul: Registering Check with ID: e7a0a5ece0c8424b3c8b06248040a331d999558b for Service: a832123c-17a8-d041-afcb-aeb29687634b-cache-redis
    2015/12/09 22:01:20 [ERROR] consul: Error while registerting check alive with consul: Put http://127.0.0.1:8500/v1/agent/check/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
    2015/12/09 22:01:25 [DEBUG] consul: Error while registering service cache-redis with consul: Put http://127.0.0.1:8500/v1/agent/service/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
    2015/12/09 22:01:25 [DEBUG] consul: Registering Check with ID: e7a0a5ece0c8424b3c8b06248040a331d999558b for Service: a832123c-17a8-d041-afcb-aeb29687634b-cache-redis
    2015/12/09 22:01:25 [ERROR] consul: Error while registerting check alive with consul: Put http://127.0.0.1:8500/v1/agent/check/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
^C==> Caught signal: interrupt
    2015/12/09 22:01:29 [DEBUG] http: Shutting down http server
    2015/12/09 22:01:29 [INFO] agent: requesting shutdown
    2015/12/09 22:01:29 [INFO] client: shutting down
2015/12/09 22:01:29 [INFO] driver.docker: stopped container 6823e10417490fa25136c2bd21b5c92a0849309bf2f3008b592d6e36f7d0c9a8
2015/12/09 22:01:29 [INFO] driver.docker: removed container 6823e10417490fa25136c2bd21b5c92a0849309bf2f3008b592d6e36f7d0c9a8
    2015/12/09 22:01:30 [DEBUG] consul: Error while registering service cache-redis with consul: Put http://127.0.0.1:8500/v1/agent/service/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
    2015/12/09 22:01:30 [DEBUG] consul: Registering Check with ID: e7a0a5ece0c8424b3c8b06248040a331d999558b for Service: a832123c-17a8-d041-afcb-aeb29687634b-cache-redis
    2015/12/09 22:01:30 [ERROR] consul: Error while registerting check alive with consul: Put http://127.0.0.1:8500/v1/agent/check/register: dial tcp 127.0.0.1:8500: getsockopt: connection refused
2015/12/09 22:01:35 [INFO] driver.docker: removed image 0643f0c7d17d348881b2dab88cfdb57ad86dcf32ea22bd0b2cd0957b6c7ab8c3
    2015/12/09 22:01:35 [INFO] consul: De-Registering service cache-redis with consul
    2015/12/09 22:01:35 [DEBUG] consul: Error in de-registering service cache-redis from consul
    2015/12/09 22:01:35 [DEBUG] client: updated allocations at index 12 (1 allocs)
    2015/12/09 22:01:35 [DEBUG] client: allocs: (added 0) (removed 0) (updated 1) (ignore 0)
    2015/12/09 22:01:35 [DEBUG] client: terminating runner for alloc 'a832123c-17a8-d041-afcb-aeb29687634b'
    2015/12/09 22:01:35 [INFO] nomad: shutting down server
    2015/12/09 22:01:35 [WARN] serf: Shutdown without a Leave
    2015/12/09 22:01:35 [INFO] consul: Shutting down consul Client
    2015/12/09 22:01:35 [INFO] agent: shutdown complete
```

**Comment by diptanu** [comment]
@cdrage Those consul errors are indicating that you don't have consul running and the job file had a service block defined so Nomad is trying to register them with Consul.

We will be muting the debug messages if consul is not present.

**Comment by cdrage** [comment]
@diptanu 
Ohhhhhh, was this changed recently? before when I had this running I was deploying it willy nilly in a container without having to have consul running, did something change in the example `./nomad init && ./nomad run example.nomad` ?

**Comment by diptanu** [comment]
@cdrage Yeah we always try to add the new features in the example, so we added the service definition block to the redis task when Consul integration was added.

**Comment by cdrage** [comment]
@diptanu ahhhhhh, is it possible if perhaps the documentation can be updated in regards to running the example? I've had a look through the website but unfortunatley couldn't find anything indicating that consul should be running before running `./nomad run example.nomad`. There's been a few blogs that have been written since 0.1 came out that basically show's the workflow as:

```bash
./nomad agent -dev -network-interface eth0
./nomad init
./nomad run example.nomad
```

which would be broken now since nomad detects that consul isn't running.

nor if I do a ctrl+f do I find anything in example.nomad about consul :(

**Comment by cdrage** [comment]
also, does this consul integration have to be ran as a cluster? (one on every node, etc.) for it to work?

**Comment by diptanu** [comment]
@cdrage Yeah for any production deployment we recommend to cluster the Consul deployments.

**Comment by diptanu** [comment]
@cdrage So we have a fix which is getting released with 0.2.2 which won't print the debug messages if Consul is not running.

However, I suggest removing the ```service``` block from your job configuration file if you're are not using Consul at all.

**Comment by cdrage** [comment]
@diptanu awesome! thanks man, much appreciated, <3 the project so far
